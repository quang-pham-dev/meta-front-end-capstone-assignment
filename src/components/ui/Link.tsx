import React from "react";
import {
  Link as RouterLink,
  LinkProps as RouterLinkProps
} from "react-router-dom";

interface LinkProps extends RouterLinkProps {
  className?: string;
  children: React.ReactNode;
}

const Link: React.FC<LinkProps> = ({ to, className, children, ...rest }) => {
  return (
    <RouterLink to={to} className={`${className || ""}`} {...rest}>
      {children}
    </RouterLink>
  );
};

Link.displayName = "Link";

export { Link };
