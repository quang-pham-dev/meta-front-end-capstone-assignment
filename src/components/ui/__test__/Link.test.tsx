import React from "react";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { BrowserRouter } from "react-router-dom";
import { Link } from "../Link";
import "@testing-library/jest-dom";

const renderWithRouter = (ui: React.ReactElement, { route = "/" } = {}) => {
  window.history.pushState({}, "Test page", route);
  return render(ui, { wrapper: BrowserRouter });
};

describe("Link", () => {
  it("renders with default props", () => {
    renderWithRouter(<Link to="/">Home</Link>);
    const link = screen.getByRole("link", { name: /home/i });
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", "/");
  });

  it("renders with custom className", () => {
    renderWithRouter(
      <Link to="/about" className="custom-link">
        About
      </Link>
    );
    const link = screen.getByRole("link", { name: /about/i });
    expect(link).toHaveClass("custom-link");
  });

  it("passes additional props to the link element", () => {
    renderWithRouter(
      <Link to="/contact" target="_blank">
        Contact
      </Link>
    );
    const link = screen.getByRole("link", { name: /contact/i });
    expect(link).toHaveAttribute("target", "_blank");
  });

  it("matches snapshot", () => {
    const { asFragment } = renderWithRouter(
      <Link to="/products">Products</Link>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
