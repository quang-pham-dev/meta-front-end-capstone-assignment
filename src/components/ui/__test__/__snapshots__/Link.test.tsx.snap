// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Link > matches snapshot 1`] = `
<DocumentFragment>
  <a
    class=""
    href="/products"
  >
    Products
  </a>
</DocumentFragment>
`;
